code-comment-style.md

```diff
@@ -0,0 +1,128 @@
+# Code Comment Style
+
+This document describes the code comment style applied to TiKV repositories. When you are to commit, be sure to follow the style to write good code comments.
+
+## Why a good comment matters?
+
+- To speed up the reviewing process
+- To help maintain the code
+- To improve the API document readability
+- To improve the development efficiency of the whole team
+
+## Where/When to comment?
+
+Write a comment where/when:
+
+- For important code
+- For obscure code
+- For tricky or interesting code
+- For a complex code block
+- If a bug exists in the code but you cannot fix it or you just want to ignore it for the moment
+- If the code is not optimal but you don’t have a smarter way now
+- To remind yourself or others of missing functionality or upcoming requirements not present in the code
+
+A comment is generally required for:
+
+- File
+- Type
+- Constant
+- Function
+- Method
+- Variable
+- Typical algorithm
+- Exported name
+- Test case

```

Comment:
My preference is not to require comments anywhere since that leads to boilerplate. Use comments when there is context that is missing from the code or is hard to deduce from the code. For example, what are the side-effects of calling a function, or the performance characteristics, or the use cases which the code addresses.

As an explicit counter-example I think comments like:

```
/// Returns a record from the database.
pub fn get_record(db: &Database) -> Record { ... }
```

are a negative and should be avoided, better to have no comment at all.

A good comment is something like:

```
/// This function may get the record from disk which means it will block waiting for IO. It is nearly always
/// better to call `async_get_record` unless you are sure that the requested record is cached in memory.
///
/// If you need to get many records, calling this function many times has O(n) performance. Consider using
/// `get_records` which has O(log n) performance.
///
/// If `db` is in an inconsistent state, this function will panic.
pub fn get_record(db: &Database) -> Record { ... }
```

Note that we don't say what `get_record` does but describe aspects of how it should be used which are not obvious from the function signature.

Reply:
Good point. Will add in style guide. We should not comment for comment purposes, but comment to deliver something that matters